src/gui/issueswidget.cpp

```diff
@@ -145,7 +146,13 @@ void IssuesWidget::hideEvent(QHideEvent *ev)
     QWidget::hideEvent(ev);
 }
 
-void IssuesWidget::cleanItems(const QString &folder)
+static bool persistsUntilLocalDiscovery(QTreeWidgetItem *item)
+{
+    const auto status = ProtocolItem::status(item);
+    return status == SyncFileItem::Conflict || status == SyncFileItem::FileIgnored;
```

Comment:
what about blacklisted errors?

I guess everything should persist.

Reply:
The concern was about error items that could appear even without a local discovery. Like a download that failed. But now I know that the same thing might happen for `FileIgnored` meaning I need a new approach.

How about I make sure that items completing successfully also wipe issue items and that there can only be one issue item per file path? I think we need to think about issue item lifetime some more...

Reply:
After more testing it became clear that the current state is also broken. If you have a remote file `foo/remote-ignore.crdownload` then this will be listed as ignored on the initial sync. But the entry is wiped in a follow up sync and does not reappear unless the folder `foo` is rediscovered.

The only correct behavior that I see is to store per entry information about when it should be wiped. `FileIgnored` would depend either on local or remote discovery (could use the `direction` tag to propagate that information).

For all issues that aren't `FileIgnored` or `Conflict`, I'd stick to the behavior of invalidating them when a new sync propagation starts - an invalidation bug that makes scary looking errors stay around when they are no longer applicable would be bad.